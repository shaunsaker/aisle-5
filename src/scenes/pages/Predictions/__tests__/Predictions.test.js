import React from 'react';
import renderer from 'react-test-renderer';

import { Predictions } from '..';
import USER_LISTS from '../../../../mockData/userLists';
import USER_ITEMS from '../../../../mockData/userItems';
import HEADER_ITEMS from '../../../../components/PredictedItemsList/headerItems';

jest.mock('../../../../components/TabBar', () => 'TabBar');

describe('Predictions', () => {
  const spies = [];
  const dispatch = jest.fn();
  const selectedHeader = HEADER_ITEMS[0];
  const { fieldName } = selectedHeader;

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <Predictions userLists={USER_LISTS} userItems={USER_ITEMS} />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle onSortByFieldName', () => {
      spies[0] = jest.spyOn(Predictions.prototype, 'setSortByKey');
      spies[1] = jest.spyOn(Predictions.prototype, 'toggleReverseSort');
      const component = renderer.create(
        <Predictions userLists={USER_LISTS} userItems={USER_ITEMS} />,
      );
      const instance = component.getInstance();

      instance.onSortByFieldName(fieldName);

      expect(spies[0]).toHaveBeenCalledWith(fieldName);

      instance.onSortByFieldName(fieldName);

      expect(spies[1]).toHaveBeenCalled();
    });

    it('should handle setSortByKey', () => {
      const component = renderer.create(
        <Predictions userLists={USER_LISTS} userItems={USER_ITEMS} />,
      );
      const instance = component.getInstance();

      instance.setSortByKey(fieldName);

      expect(instance.state.sortByKey).toEqual(fieldName);
    });

    it('should handle toggleReverseSort', () => {
      const component = renderer.create(
        <Predictions userLists={USER_LISTS} userItems={USER_ITEMS} />,
      );
      const instance = component.getInstance();

      expect(instance.state.reverseSort).toEqual(false); // initial state

      instance.toggleReverseSort();

      expect(instance.state.reverseSort).toEqual(true);
    });
  });

  describe('actions', () => {
    it('should call onSortByFieldName on header item press', () => {
      const component = renderer.create(
        <Predictions userLists={USER_LISTS} userItems={USER_ITEMS} />,
      );
      const { root } = component;
      const targetComponent = root.findByProps({
        testID: `predictions.predictedItemsList.header.button.${selectedHeader.text}`,
      });

      targetComponent.props.onPress();

      expect(spies[0]).toHaveBeenCalledWith(fieldName);
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
