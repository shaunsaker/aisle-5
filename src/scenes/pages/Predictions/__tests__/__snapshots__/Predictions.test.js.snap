// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Predictions renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#EDE8CA",
        "flex": 1,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#08293A",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#08293A",
          "borderColor": null,
          "borderWidth": 0,
          "elevation": 12,
          "height": 60,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingHorizontal": 15,
          "paddingTop": 10,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 4,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#212121",
            "fontFamily": "ArchivoBlack-Regular",
            "fontSize": 24,
          },
          Object {
            "color": "#EDE8CA",
          },
        ]
      }
    >
      Predictions
    </Text>
    <View
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <NetworkActivityIndicator />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={[Function]}
      contentContainerStyle={Object {}}
      data={
        Array [
          Object {
            "averageUsagePerDay": 5184,
            "daysLeftUntilZeroQuantity": 0.0023148148148148147,
            "id": "1",
            "name": "Beer",
            "previouslyPurchasedQuantity": 6,
            "remainingQuantity": 12,
            "remainingQuantityPercentage": 200,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      style={Object {}}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EDE8CA",
                  "borderBottomColor": "#FFFFFF",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 15,
                  "paddingVertical": 10,
                },
                Object {
                  "height": 51,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "marginRight": 15,
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="predictions.predictedItemsList.header.button.NAME"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#08293A",
                        "fontFamily": "Muli-Bold",
                        "fontSize": 15,
                        "textDecorationLine": "underline",
                      },
                      undefined,
                    ]
                  }
                >
                  NAME
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#212121",
                      "fontSize": 24,
                      "marginLeft": 7.5,
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "flex": 2,
                  },
                  Object {
                    "marginRight": 0,
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="predictions.predictedItemsList.header.button.CONSUMPTION"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#08293A",
                        "fontFamily": "Muli-Bold",
                        "fontSize": 15,
                        "textDecorationLine": "underline",
                      },
                      undefined,
                    ]
                  }
                >
                  CONSUMPTION
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#212121",
                      "fontSize": 24,
                      "marginLeft": 7.5,
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={null}
        >
          <View
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  },
                  Object {
                    "height": 51,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "marginRight": 15,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      Object {
                        "height": undefined,
                      },
                    ]
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 20,
                        }
                      }
                    >
                      🍺
                       
                    </Text>
                  </View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "#212121",
                        "flex": 1,
                        "fontFamily": "Muli-SemiBold",
                        "fontSize": 17,
                      }
                    }
                  >
                    Beer
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "stretch",
                      "backgroundColor": "rgba(42, 213, 42, 1)",
                      "borderRadius": 5,
                      "flex": 1,
                      "padding": 7.5,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#EDE8CA",
                        "borderColor": null,
                        "borderRadius": 5,
                        "borderWidth": 0,
                        "bottom": 5,
                        "elevation": 2,
                        "left": 7.5,
                        "position": "absolute",
                        "right": 7.5,
                        "shadowColor": "#212121",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.17,
                        "shadowRadius": 2,
                        "top": 5,
                        "width": "100%",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "elevation": 100,
                      "justifyContent": "center",
                      "position": "absolute",
                      "right": 15,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#212121",
                        "fontFamily": "Muli-SemiBold",
                        "fontSize": 15,
                      }
                    }
                  >
                    0 days left
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <TabBar />
</View>
`;
